[WIP] IBM Power VS Block CSI Driver Operator (#54711)
openshift/openshift-docs: storage/container_storage_interface/persistent-storage-csi-ibm-powervs-block.adoc

Reviewer: The assembly needs an introduction: https://redhat-documentation.github.io/modular-docs/#assembly-guidelines

Reviewer 2: This documentation follows the same format as every other csi driver operator in the OpenShift docs.
Ref: https://docs.openshift.com/container-platform/4.8/storage/container_storage_interface/persistent-storage-csi-gcp-pd.html

Do we need to put additional introduction info?

Reviewer 2: cc: @maxwelldb

Reviewer: @Madhan-SWE I'd say that those docs should also follow the format in our guidelines. Can't explain why they don't. There could be an exception for these assemblies, but I don't have a great way to find that out.

Reviewer 3: I'm not aware of an exception - assemblies need an introduction so that helpful text is more likely to show up in search results and users can quickly determine if they've found the topic that they need. My suspicion is that it slipped through review on the [first assembly](https://github.com/openshift/openshift-docs/pull/22199) like this and was just copied over and over again.

I suggest adding an introduction and filing an issue in OSDOCS to update the other similar assemblies.

Reviewer 2: **Introduction**

The [IBM Power Virtual Server Block CSI Driver](https://github.com/openshift/ibm-powervs-block-csi-driver) will be installed via [IBM Power Virtual Server Block CSI Driver Operator](https://github.com/openshift/ibm-powervs-block-csi-driver-operator) and the operator is based on [libarary-go](https://github.com/openshift/library-go/). openshift/library-go is a collection of functions that allow us to build OpenShift operators easily. Most of the functionality of a CSI driver operator is already available there.The IBM Power Virtual Server Block CSI Driver Operator is installed by the [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator). Cluster-storage-operator installs the IBM Power Virtual Server Block CSI Driver Operator if the Platform type is Power Virtual Servers. The CSI onboarding guide for Openstack Infrastructure Provider can be found [here](https://docs.providers.openshift.org/container-storage-interface-driver/).

Reviewer 2: cc: @maxwelldb @alishaIBM

Reviewer 2: Please review if this looks good. We can add it in the docs.

Reviewer 3: @sferich888, since these are all within the openshift org, are you ok with them? Or are there specific requirements you'd like to be confirmed first?

Reviewer 4: We should not link to GitHub no matter where the content is. 

If we are providing a link for a purpose then the content of that link should be documented in our docs. 

Most of these looks like references (to whit I have to ask the value of the reference), shot of providing a breadcrumb trail.

Author: Added new content (without GitHub links).

Author: @kelbrown20 @kalexand-rh could you please review the PR content ? Thanks.

[id="persistent-storage-csi-ibm-powervs-block"]
= IBM Power VS Block CSI Driver Operator
include::_attributes/common-attributes.adoc[]
:context: persistent-storage-csi-ibm-powervs-block

toc::[]

== Overview

> {product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for IBM Power VS Block Storage.

Reviewer 2: ```suggestion
{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for IBM PowerVS Block Storage.
```

> 

Reviewer 5: I was testing some formating and I don't think there needs to be a space between these two paragraphs

Author: @kelbrown20 , I have referred to this module https://54711--docspreview.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-ibm-vpc-block.html#overview which use similar format. 
Request you to please check above once and suggest on the same. Thanks.

Reviewer 5: Oh gotcha! Yeah that looks good to me then!

Author: Thanks @kelbrown20 . Marking the comment as resolved.

Familiarity with xref:../../storage/understanding-persistent-storage.adoc#understanding-persistent-storage[persistent storage] and xref:../../storage/container_storage_interface/persistent-storage-csi.adoc#persistent-storage-csi[configuring CSI volumes] is recommended when working with a CSI Operator and driver.

> To create CSI-provisioned PVs that mount to IBM Power VS Block storage assets, {product-title} installs the IBM Power VS Block CSI Driver Operator and the IBM VPC Block CSI driver by default in the `openshift-cluster-csi-drivers` namespace.

Reviewer 2: ```suggestion
To create CSI-provisioned PVs that mount to IBM Power VS Block storage assets, {product-title} installs the IBM Power VS Block CSI Driver Operator and the IBM Power VS Block CSI driver by default in the `openshift-cluster-csi-drivers` namespace.
```

Author: Ack.

> To create CSI-provisioned PVs that mount to IBM Power VS Block storage assets, {product-title} installs the IBM Power VS Block CSI Driver Operator and the IBM Power VS Block CSI driver by default in the `openshift-cluster-csi-drivers` namespace.

Reviewer 2: ```suggestion
To create CSI-provisioned PVs that mount to IBM PowerVS Block storage assets, {product-title} installs the IBM PowerVS Block CSI Driver Operator and the IBM PowerVS Block CSI driver by default in the `openshift-cluster-csi-drivers` namespace.
```

> * The _IBM Power VS Block CSI Driver Operator_ provides two storage classes named `ibm-powervs-tier1` (default), and `ibm-powervs-tier3` for different tiers that you can use to create persistent volume claims (PVCs). The IBM Power VS Block CSI Driver Operator supports dynamic volume provisioning by allowing storage volumes to be created on demand, eliminating the need for cluster administrators to pre-provision storage.

Reviewer 2: ```suggestion
* The _IBM PowerVS Block CSI Driver Operator_ provides two storage classes named `ibm-powervs-tier1` (default), and `ibm-powervs-tier3` for different tiers that you can use to create persistent volume claims (PVCs). The IBM PowerVS Block CSI Driver Operator supports dynamic volume provisioning by allowing storage volumes to be created on demand, eliminating the need for cluster administrators to pre-provision storage.
```

> * The _IBM Power VS Block CSI driver_ enables you to create and mount IBM Power VS Block PVs.

Reviewer 2: ```suggestion
* The _IBM PowerVS Block CSI driver_ enables you to create and mount IBM PowerVS Block PVs.
```

> * The _{ibmpowerProductName} Virtual Server Block CSI Driver Operator_ provides two storage classes named `ibm-powervs-tier1` (default), and `ibm-powervs-tier3` for different tiers that you can use to create persistent volume claims (PVCs). The {ibmpowerProductName} Virtual Server Block CSI Driver Operator supports dynamic volume provisioning by allowing storage volumes to be created on demand, eliminating the need for cluster administrators to pre-provision storage.

Reviewer 5: I think I understand why you put bullet points here, but I'm still not sure if they are necessary. It seems like most bullet points used in Overviews are used when listing something.

Author: @kelbrown20 , same as above. I have referred to this module https://54711--docspreview.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-ibm-vpc-block.html#overview which use similar format.
Request you to please check above once and suggest on the same. Thanks.

Reviewer 5: Gotcha! Okay, yeah that looks good to me then!

Author: Thanks @kelbrown20. Marking the comment as resolved.

[id="persistent-storage-csi-ibm-powervs-block"]
= {ibmpowerProductName} Virtual Server Block CSI Driver Operator
include::_attributes/common-attributes.adoc[]
:context: persistent-storage-csi-ibm-powervs-block

toc::[]

== Overview

> {product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for {ibmpowerProductName} Virtual Server Block Storage.

Reviewer: ISG suggestion:

```suggestion
{product-title} can provision persistent volumes (PVs) by using the Container Storage Interface (CSI) driver for {ibmpowerProductName} Virtual Server Block Storage.
```

Author: Thanks. Updated.

:FeatureName: {ibmpowerProductName} Virtual Server Block CSI Driver Operator

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: Added Technology Preview Note. FYI : @manojnkumar @Madhan-SWE @kelbrown20

> Familiarity with xref:../../storage/understanding-persistent-storage.adoc#understanding-persistent-storage[persistent storage] and xref:../../storage/container_storage_interface/persistent-storage-csi.adoc#persistent-storage-csi[configuring CSI volumes] is recommended when working with a CSI Operator and driver.

Reviewer: :warning: ISG: Could revise to eliminate passive voice and use of "recommended."

To create CSI-provisioned PVs that mount to {ibmpowerProductName} Virtual Server Block storage assets, {product-title} installs the {ibmpowerProductName} Virtual Server Block CSI Driver Operator and the {ibmpowerProductName} Virtual Server Block CSI driver by default in the `openshift-cluster-csi-drivers` namespace.

* The _{ibmpowerProductName} Virtual Server Block CSI Driver Operator_ provides two storage classes named `ibm-powervs-tier1` (default), and `ibm-powervs-tier3` for different tiers that you can use to create persistent volume claims (PVCs). The {ibmpowerProductName} Virtual Server Block CSI Driver Operator supports dynamic volume provisioning by allowing storage volumes to be created on demand, eliminating the need for cluster administrators to pre-provision storage.

> * The _{ibmpowerProductName} Virtual Server Block CSI driver_ enables you to create and mount {ibmpowerProductName} Virtual Server Block PVs.

Reviewer: :warning: ISG: Anthropomorphization in "enables".

[id="persistent-storage-csi-ibm-powervs-block"]
= {ibmpowerProductName} Virtual Server Block CSI Driver Operator
include::_attributes/common-attributes.adoc[]
:context: persistent-storage-csi-ibm-powervs-block

toc::[]

> :_content-type: CONCEPT

Reviewer 6: According to our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata), the `content-type` needs to be on line 1. Also, this needs to be `ASSEMBLY` not `CONCEPT`.

Author: removed : 
:_content-type: CONCEPT

@sheriff-rh please check if correct. Thanks.

Reviewer 6: @alishaIBM content-type needs to be on line 1 and it needs to be ASSEMBLY type: 

```
:_content-type: ASSEMBLY
[id="persistent-storage-csi-ibm-powervs-block"]
= {ibmpowerProductName} Virtual Server Block CSI Driver Operator
include::_attributes/common-attributes.adoc[]
:context: persistent-storage-csi-ibm-powervs-block

toc::[]
```

Author: @sheriff-rh 
Incorporated above suggestion. Thanks.

[id="persistent-storage-csi-ibm-powervs-block-overview_{context}"]
== Introduction

The IBM Power Virtual Server Block CSI Driver will be installed through IBM Power Virtual Server Block CSI Driver Operator and the operator is based on libarary-go. The OpenShift library-go is a collection of functions that allow us to build OpenShift operators easily. Most of the functionality of a CSI driver operator is already available there. The IBM Power Virtual Server Block CSI Driver Operator is installed by the cluster-storage-operator. The Cluster-storage-operator installs the IBM Power Virtual Server Block CSI Driver Operator if the Platform type is Power Virtual Servers.

> == Overview

Reviewer 6: This needs an `[id=]` block before Overview.

Author: Added an [id=] block before Overview.
